app/views/handlers/list_edit_question.py

```diff
@@ -13,6 +15,31 @@ def is_location_valid(self):
             return False
         return True
 
+    def get_next_location_url(self):
+        """
+        Unless editing from the summary page, If there are repeating blocks and not all are complete, go to the next one
+        """
+        if self._is_returning_to_section_summary():
```

Comment:
Yep good point - if you use `test_list_collector_repeating_blocks_section_summary` and go to the section-summary, then edit a list item or repeating block answer, previous takes you to the summary

if you do it from the /submit/ page and use previous, you go back to the section summary rather than the /submit/ page as the url had the wrong return to location

Now it exits out to the list collector, because final-summary doesn't have the same handling, will have a look at generalising that to other types of summary

Reply:
Now fixed, test added